src/api/c/plot.cpp

```diff
@@ -64,6 +64,43 @@ forge::Chart* setup_plot(const forge::Window* const window, const af_array in_,
     // ArrayFire LOGO Orange shade
     plot->setColor(0.929f, 0.529f, 0.212f, 1.0);
 
+    // If chart axes limits do not have a manual override
+    // then compute and set axes limits
+    if(!fgMngr.getChartAxesOverride(chart)) {
+        float cmin[3], cmax[3];
+        T     dmin[3], dmax[3];
+        chart->getAxesLimits(&cmin[0], &cmax[0], &cmin[1], &cmax[1], &cmin[2], &cmax[2]);
+        copyData(dmin, reduce<af_min_t, T, T>(in, 1));
+        copyData(dmax, reduce<af_max_t, T, T>(in, 1));
+
+        if(cmin[0] == 0 && cmax[0] == 0
+        && cmin[1] == 0 && cmax[1] == 0
+        && cmin[2] == 0 && cmax[2] == 0) {
+            // No previous limits. Set without checking
+            cmin[0] = step_round(dmin[0], false);
+            cmax[0] = step_round(dmax[0], true);
+            cmin[1] = step_round(dmin[1], false);
+            cmax[1] = step_round(dmax[1], true);
+            if(order == 3) cmin[2] = step_round(dmin[2], false);
+            if(order == 3) cmax[2] = step_round(dmax[2], true);
+        } else {
+            if(cmin[0] > dmin[0])       cmin[0] = step_round(dmin[0], false);
```

Comment:
This is basically to ensure that the limits can only grow and not shrink. So minimum can only become lesser and maximum can only become greater.